tools/riemann-elasticsearch/bin/riemann-elasticsearch

```diff
@@ -35,90 +36,122 @@ class Riemann::Tools::Elasticsearch
       response
   end
 
-  def health_url
+  def make_es_url(path)
     path_prefix = options[:path_prefix]
     path_prefix[0] = '' if path_prefix[0]=='/'
     path_prefix[path_prefix.length-1] = '' if path_prefix[path_prefix.length-1]=='/'
-    "http://#{options[:es_host]}:#{options[:es_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/_cluster/health"
+    "http://#{options[:es_host]}:#{options[:es_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/#{path}"
+  end
+
+  def health_url
+    make_es_url("_cluster/health")
   end
 
   def indices_url
-    path_prefix = options[:path_prefix]
-    path_prefix[0] = '' if path_prefix[0]=='/'
-    path_prefix[path_prefix.length-1] = '' if path_prefix[path_prefix.length-1]=='/'
-    "http://#{options[:es_host]}:#{options[:es_port]}#{path_prefix.length>0?'/':''}#{path_prefix}/_stats/index,store"
+    make_es_url("_stats/index,store")
+  end
+
+  def search_url
+    es_search_index = options[:es_search_index]
+    make_es_url("#{es_search_index}/_stats/search")
+  end
+
+  def response_is_bad(response)
```

Comment:
This could be turned into:

```ruby
def response_is_bad?(response)

... 

```

Minus the return because the if should return `true`.